satpy/tests/reader_tests/test_seviri_l1b_native.py

```diff
@@ -1248,7 +1250,127 @@ def test_read_header():
+    """
+
+    def _amend_15_MAIN_PRODUCT_HEADER():
+        hdr_null_numpy[0][0][0] = (b"FormatName                  : ", b"NATIVE\n")
+
+    def _amend_15_SECONDARY_PRODUCT_HEADER():
+        hdr_null_numpy[0][1][9] = (b"SelectedBandIDs", b"XXXXXXXXXXXX")
+        hdr_null_numpy[0][1][10] = (b"SouthLineSelectedRectangle", b"1")
+        hdr_null_numpy[0][1][11] = (b"NorthLineSelectedRectangle", b"3712")
+        hdr_null_numpy[0][1][12] = (b"EastColumnSelectedRectangle", b"1")
+        hdr_null_numpy[0][1][13] = (b"WestColumnSelectedRectangle", b"3712")
+        hdr_null_numpy[0][1][14] = (b"NumberLinesVISIR", b"3712")
+        hdr_null_numpy[0][1][15] = (b"NumberColumnsVISIR", b"3712")
+        hdr_null_numpy[0][1][16] = (b"NumberLinesHRV", b"11136")
+        hdr_null_numpy[0][1][17] = (b"NumberColumnsHRV", b"11136")
+
+    def _amend_GP_PK_SH1__PacketTime():
+        hdr_null_numpy[0][3][5] = (23158, 27921912)
+
+    def _amend_15_DATA_HEADER__SatelliteStatus__SatelliteDefinition__SatelliteId():
+        hdr_null_numpy[0][4][1][0][0] = 324
+
+    def _amend_15_DATA_HEADER__GeometricProcessing__EarthModel():
+        hdr_null_numpy[0][4][6][1] = (2, 6378.169, 6356.5838, 6356.5838)
+
+    def _amend_15_DATA_HEADER__ImageAcquisition__PlannedAcquisitionTime():
+        hdr_null_numpy[0][4][2][0] = (
+            (23158, 27911177, 286, 223),
+            (23158, 28663675, 401, 687),
+            (23158, 28810078, 157, 663)
+        )
+
+    # Apply all the header amendments
+    _amend_15_MAIN_PRODUCT_HEADER()
+    _amend_15_SECONDARY_PRODUCT_HEADER()
+    _amend_GP_PK_SH1__PacketTime()
+    _amend_15_DATA_HEADER__SatelliteStatus__SatelliteDefinition__SatelliteId()
+    _amend_15_DATA_HEADER__GeometricProcessing__EarthModel()
+    _amend_15_DATA_HEADER__ImageAcquisition__PlannedAcquisitionTime()
+
+
+@pytest.fixture()
+def tmp_seviri_nat_filename(tmp_path):
+    """Creates a fully-qualified filename for a seviri native format file."""
+    tmp_filename = "MSG4-SEVI-MSG15-0100-NA-20210528075743.722000000Z-NA"
+    return tmp_path / f"{tmp_filename}.nat"
+
+
+def append_data_and_trailer_content_to_seviri_native_header(filename, hdr_null_numpy):
+    """Generates the data and trailer part (null content) of the file and appends them to the null header.
+
+    The data and trailer are also null and appending them to the header results in a complete seviri nat file.
+    """
+    # size of different parts of the seviri native file in bytes
+    size = dict(header_with_archive=450400, data=270344960, trailer=380363)
+
+    zero_bytes = bytearray(size["data"] + size["trailer"])
+    bytes_data = bytes(zero_bytes)
+
+    hdr_null_numpy.tofile(filename)
+    with open(filename, "ab") as f:
+        f.write(bytes_data)
+
+
+@pytest.fixture()
+def physical_seviri_native_file(tmp_seviri_nat_filename):
+    """Creates a physical seviri native file on disk."""
```

Comment:
would it be possible to invert the order of the functions to that the main function (this fixture) would be above its dependencies? I think it makes it more readable.